src/rust/bitbox02-rust/src/async_usb.rs

```diff
@@ -29,20 +29,20 @@ type UsbIn = Vec<u8>;
 /// Describes the global state of an api query. The documentation of
 /// the variants apply to the HWW stack, but have analogous meaning in
 /// the U2F stack.
-enum UsbTaskState {
+enum UsbTaskState<'a> {
     /// Waiting for a new query, nothing to do.
     Nothing,
     /// A query came in which launched a task, which is now running (e.g. user is entering a
     /// password).
-    Running(Task<UsbOut>),
+    Running(Task<'a, UsbOut>),
     /// The task has finished and written the result, so the USB response is available. We are now
     /// waiting for the client to fetch it (HWW_REQ_RETRY). For short-circuited or non-async api
     /// calls, the result might be returned immediately in response to HWW_REQ_NEW.
     ResultAvailable(UsbOut),
 }
 
 /// A safer version of UsbTaskState. RefCell so we cannot accidentally borrow illegally.
-struct SafeUsbTaskState(RefCell<UsbTaskState>);
+struct SafeUsbTaskState(RefCell<UsbTaskState<'static>>);
```

Comment:
Why does it have to be static?

Reply:
a few lines down the only use of this is in a static variable. I could make this generic too and have the compiler infer it (I guess!), but that would clutter it more with generic params.

Reply:
ah no, this is good, sorry. I missed the static below. should've unfolded more lines.
:+1: